lms/services/application_instance.py

```diff
@@ -58,6 +58,27 @@ def get_by_consumer_key(self, consumer_key) -> ApplicationInstance:
         except NoResultFound as err:
             raise ApplicationInstanceNotFound() from err
 
+    @lru_cache
+    def get_by_deployment_id(
```

Comment:
I think this is closer to being right as it is actually. I think 1.3 is super vague.

What I'm less sure is if this is looking it up by deployment any more than it's looking it up by any of the other things. I think:

* If there's a collective noun of some kind for `issuer`, `client_id` and `deployment_id` - then that's best
* If not, then I think the main thing this name does for us is disinguish itself from other things and make it greppable... which this totally does

Reply:
I think the name for `issuer + client_id + deployment_id` is "install" or application_instance itself.

I leave it as get_by_deployment_id.

Reply:
Maybe this should be ```get_blah(lti_v13_params: LTI13Params)``` then? Not sure how to name that, or if that's much worse.

Reply:
Not sure, the only moment we use this is the callback of the auth flow, before an actual LTI launch is made so that's why I preferred the separated params.

I can't find a good name for them that is not a synonym of `application_instance`.

Reply:
Ignore me! This isn't used in an LTI "launch" context at all, and is called with a custom schema elsewhere.